ci/ray_ci/doc/api.py

```diff
@@ -19,3 +26,64 @@ class API:
     name: str
     annotation_type: AnnotationType
     code_type: CodeType
+
+    def from_autosummary(doc: str, current_module: Optional[str] = None):
+        """
+        Parse API from the following autosummary sphinx block.
+
+        .. autosummary::
+            :option_01
+            :option_02
+
+            api_01
+            api_02
+        """
+        apis = []
+        lines = doc.splitlines()
+        if not lines:
+            return apis
+
+        if lines[0].strip() != SPHINX_AUTOSUMMARY_HEADER:
+            return apis
+
+        for line in lines:
+            if line == SPHINX_AUTOSUMMARY_HEADER:
```

Comment:
do we want to just `strip()` before comparison? also, the line is being striped many times afterwards.

Reply:
not really, the re.match(r"\s", line) check if the line is not empty and not starting with empty space, so we want to keep the space if those exists